### ISSUE TITLE
A11y keyboard focus control in the SnackBar Component

### ISSUE THREAD

**Issue by DaveBest99** [angular/components#3773]
#### Bug, feature request, or proposal:
Bug: The aria-alert in the "Example" Tab does not work as expected when using Mozilla Firefox or Google Chrome.

#### What is the expected behavior?
Message should be automatically spoken.

#### What is the current behavior?
Currently in Firefox and Chrome the screen reader does not announce the message.

#### What are the steps to reproduce?
See below.

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

#### What is the use-case or motivation for changing an existing behavior?
Create a good screen reader user experience.

#### Which versions of Angular, Material, OS, browsers are affected?
Using screen reader JAWS17 with browsers Firefox, Internet Explorer 11, and Google Chrome, on Windows10.

#### Is there anything else we should know?
Screen reader user experience follows:

1. Press Tab key to the "nack-bar with a custom component" demo component in the "Example" Tab Panel.
JAWS says "Show an example snack-bar button", and the braille display shows 
"Show an example snack-bar btn".
- Using Internet Explorer 11: Pressing Spacebar or Enter key on the button will result in the screen reader announcing "alert" and the message, as expected.
- Using Chrome or Firefox: Pressing Spacebar or Enter key on the button will result in no screen reader action (the message is displayed at the bottom of the page but not announced by JAWS).
- Note, the test example in the "Overview" Tab works for IE11, FF, and Chrome. No idea why the "Overview" test works but not the "Example" test, when using FF or Chrome.

2. If the user, as in the demo, can press the button to repeat the message, then an aria-alert works good. However, if the message is displayed, and announced by the screen reader, but the user cannot take action to have the message repeated, then a Dialog modal with a Close button should be used, to ensure the user perceives the message.
